apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,443 @@
+    return False
+
+  def prepareResult(self, is_recursive, is_directory, is_json, alt):
+    if is_recursive:
+      return self.prepareRecursiveResult(is_json)
+
+    if alt and self.isDirectory() and self.canRequestChildren():
+      return (None, bad_request_error)
+  
+    return self.prepareNonRecursiveResult(is_json)
+
+  def prepareNonRecursiveResult(self, is_json):
+    return (unicode(self._value), None)
+
+  def prepareRecursiveResult(self, is_json):
+    if is_json:
+      return (json.dumps(self.getValueToJSONEncode(), separators=(u',', u':')), None)
+    else:
+      return (self.prepareRecursiveResultAsText(''), None)
+
+  def prepareRecursiveResultAsText(self, prefix):
+    result = unicode(self._value)
+    return prefix + u' ' + result + u'\n' if prefix else result
+
+  def value(self):
+    return self._value
+
+# Base class for a wrapper child contains child metadata values.
+class MetadataDirectory(MetadataValue):
+  def __iter__(self):
+    return self._value.__iter__()
+
+  def __len__(self):
+    return self._value.__len__()
+
+  def __delitem__(self, key):
+    self._value.__delitem__(key)
+
+  def __getitem__(self, key):
+    return self._value.__getitem__(key)
+  
+  def __setitem__(self, key, value):
+    self._value.__setitem__(key, value)
+
+  def canRequestChildren(self):
+    return True
+
+  def isDirectory(self):
+    return True
+
+# Wrapper for a set of metadata key/value pairs.
+class MetadataDict(MetadataDirectory):
+  def __init__(self, items):
+    assert isinstance(items, OrderedDict)
+    MetadataDirectory.__init__(self, items)
+
+  def getChild(self, child_name):
+    if not child_name in self:
+      return None
+
+    return self[child_name]
+
+  def getValueToJSONEncode(self):
+    result = OrderedDict()
+    for key in self:
+      if excludeKeyInRecursiveGet(key):
+        continue
+
+      if key.endswith(u'@developer.gserviceaccount.com'):
+        newKey = key
+      else:
+        newKey = common.convertToCamelCase(key)
+
+      result[newKey] = self[key].getValueToJSONEncode()
+
+    return result
+
+  def prepareNonRecursiveResult(self, is_json):
+    def includeSlash(key):
+      return not excludeKeyInRecursiveGet(key) and self[key].isDirectory() and self[key].canRequestChildren()
+
+    return (''.join(key + (u'/\n' if includeSlash(key) else u'\n') for key in self), None)
+
+  def prepareRecursiveResultAsText(self, prefix):
+    result = u''
+    for key in self:
+      if excludeKeyInRecursiveGet(key):
+        continue
+
+      result += self[key].prepareRecursiveResultAsText(prefix + (u'/' if len(prefix) != 0 else u'') + key)
+
+    return result
+
+  def value(self):
+    return self
+
+# Wrapper for a collection whose content is accessed via an index value in the relative URL.
+class MetadataIndexedList(MetadataDirectory):
+  def __init__(self, items):
+    assert isinstance(items, list)
+    MetadataDirectory.__init__(self, items)
+
+  def getChild(self, child_name):
```

Comment:
No, these are taken from the split call (on the relative path). Also, those with 0 length are filtered out, so if one was ever None somehow, it would crash on that check first anyway, but I don't think this can happen.